modules/series-service-impl/src/main/java/org/opencastproject/series/impl/persistence/SeriesEntity.java

```diff
@@ -49,17 +53,39 @@
  * rules.
  *
  */
-@Entity(name = "SeriesEntity") @IdClass(SeriesEntityId.class)
+@Entity(name = "SeriesEntity")
+@IdClass(SeriesEntityId.class)
 @Access(AccessType.FIELD)
-@Table(name = "oc_series")
+@Table(name = "oc_series",
+    indexes = {
+        @Index(name = "IX_oc_series_modified_date", columnList = ("modified_date")),
+    }
+)
 @NamedQueries({
-    @NamedQuery(name = "Series.findAll", query = "select s from SeriesEntity s"),
-    @NamedQuery(name = "Series.getCount", query = "select COUNT(s) from SeriesEntity s"),
+    @NamedQuery(
+        name = "Series.findAll",
+        query = "select s from SeriesEntity s where s.deletionDate is null"
```

Comment:
Could you explain to me when do we have a series that is deleted still in the db? As far as I know, deleted series are removed from the db, so storing a deletionDate makes little sense?

Reply:
This PR also changes that. When deleting a series, it is not actually deleted from the DB, but we only set the deleted date to a non-null value.

```diff
- em.remove(entity);
+ entity.setModifiedDate(new Date());
+ entity.setDeletionDate(new Date());
+ em.merge(entity);
```

https://github.com/opencast/opencast/pull/3076/commits/afb82d27d61370bb2d5903e4a25881c9aad11401#diff-8ebbecff33097385ffb1bdcc22e6a8acd012f150472562fa17ab98838c7afd43L172-L173

Reply:
Right, not sure how I missed that. Thanks for the link

Reply:
Does that mean there is no way of permanently deleting a series from the db anymore short of directly removing them from the db by hand?

Reply:
Yes, that is correct. I mean, for Opencast, it is deleted as all SQL queries retrieving series now explicitly check for `deletion_date is null`. But yes, the data is still in the database. 

Maybe instead of just setting the modification and deletion dates on removal, all other columns should be set to empty or dummy values? But that could always be changed later I guess?